New rodent categories with toggle working

stijnhofhuis opened the pull request:
Hei Chloe, it works now.
The only error I get is "Error: Failed to create the shared library. Run 'printErrors()' to see the compilation errors."

I thought this may not be related to anything I did?

stijnhofhuis committed:
Start at making new rodent categories

stijnhofhuis committed:
6 different rodent variables by start_hunting year

(fall / winter,
standardised for species / only for seasons,
categorical / continuous)

stijnhofhuis committed:
Fall rodent now based on storskala, winter on varanger

stijnhofhuis committed:
Making sure new rodent cat names works in model +toggle species standar.

ChloeRN commented:
@stijnhofhuis the shared library error is (unfortunately) a recurring issue when running nimble. It typically happens either because the correct version of Rtools is not installed, or because some paths are wrong.

If you haven't already, make sure to install the correct version of RTools: 
https://cran.r-project.org/bin/windows/Rtools/

If the problem persists after that, let me know.

ChloeRN commented:
@stijnhofhuis I can see that there are no NA  values now in rodent covariates starting from year (start hunting year) 2004. Is that correct?

And just to clarify: 
The rodent covariates with "start hunting year" = 2004 (for example) are now fall 2004 + spring 2005, right? So they would matter for a) survival / immigration from spring 2004 to spring 2005 and b) for reproduction in spring 2005?

ChloeRN committed:
Small edits to Stijn's code

Mostly cosmetics.

ChloeRN committed:
Changed levels of rodent index from 0-1 to 1-2

It makes coding of imputation of missing values a bit neater.

ChloeRN committed:
Adjusted time index of rodent covariates

I've moved the time index of rodent covariates forrward to a) represent that reproduction depends on previous rodent abundance and b) make it possible to impute the first covariate value. 

I have also added information on the timing to the outpug of wrangleData_rodent()

ChloeRN committed:
Moved rodent covariates from constants to data

ChloeRN committed:
Wrote imputation module for rodent covariates (missing values)

ChloeRN committed:
Added initial values for missing rodent covariate values

ChloeRN committed:
Testing & minor cosmetics

ChloeRN reviewed (approved):
I've gone over all the code now and tested running the model with different toggle combinations. 
I have made a few notable changes: 

- Changed the rodent Index levels from 0-1 to 1-2 (allows neater coding)
- Shifted the time indices of rodent covariates forward to represent that reproduction depends on prior rodent abundance (and allow imputation of values in first year)
- Added functionality for imputing missing covariate values

Everything works as per now.
